src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -506,172 +507,70 @@ namespace AppInstaller::CLI::Workflow
-                    {
-                        packagesInBoth.emplace_back(change.Package);
-                        break;
-                    }
+                    AppsAndFeaturesEntry entry = baseEntry;
+                    entry.DisplayName = std::move(names[i]).get();
+                    entry.Publisher = std::move(publishers[i]).get();
+                    entries.push_back(std::move(entry));
                 }
             }
 
-            // We now have all of the package changes; time to report them.
-            // The set of cases we could have for changes to ARP:
-            //  0 packages  ::  No changes were detected to ARP, which could mean that the installer
-            //                  did not write an entry. It could also be a forced reinstall.
-            //  1 package   ::  Golden path; this should be what we installed.
-            //  2+ packages ::  We need to determine which package actually matches the one that we
-            //                  were installing.
-            //
-            // The set of cases we could have for finding packages based on the manifest:
-            //  0 packages  ::  The manifest data does not match the ARP information.
-            //  1 package   ::  Golden path; this should be what we installed.
-            //  2+ packages ::  The data in the manifest is either too broad or we have
-            //                  a problem with our name normalization.
-
-            // Find the package that we are going to log
-            std::shared_ptr<IPackageVersion> toLog;
-
-            // If there is only a single common package (changed and matches), it is almost certainly the correct one.
-            if (packagesInBoth.size() == 1)
-            {
-                toLog = packagesInBoth[0]->GetInstalledVersion();
-            }
-            // If it wasn't changed but we still find a match, that is the best thing to report.
-            else if (findByManifest.Matches.size() == 1)
-            {
-                toLog = findByManifest.Matches[0].Package->GetInstalledVersion();
-            }
-            // If only a single ARP entry was changed and we found no matches, report that.
-            else if (findByManifest.Matches.empty() && changes.size() == 1)
-            {
-                toLog = changes[0].Package->GetInstalledVersion();
-            }
-
-            IPackageVersion::Metadata toLogMetadata;
-            if (toLog)
-            {
-                toLogMetadata = toLog->GetMetadata();
-            }
-
-            // We can only get the source identifier from an active source
-            std::string sourceIdentifier;
-            if (context.Contains(Execution::Data::PackageVersion))
-            {
-                sourceIdentifier = context.Get<Execution::Data::PackageVersion>()->GetProperty(PackageVersionProperty::SourceIdentifier);
-            }
-
-            Logging::Telemetry().LogSuccessfulInstallARPChange(

```

Comment:
I had moved it down to the function doing the correlation; but now it's back here. I changed the helper to return the count of changes/matches, although I'm keeping that count to only consider the exact matches from the source search as I couldn't figure out a good way to keep the count consistent across the multiple "passes".

Do you have any ideas how to count the matching manifests when sometimes we use the exact matching and sometimes the confidence measures?

Reply:
As long as we can reason about the meaning, anything is fine.  It can stay using the exact same values as before, just with a better guess.  I thought we might use these values in some way to find things that weren't correlating, but it turned out to be very easy to find them 😉 

So basically, these numbers are probably not important.  Don't spend time trying to improve them, and if you think they are broken, we might consider just reporting 0 for all of them.